tests/test_preload_script.py

```diff
@@ -636,3 +636,57 @@ async def test_removePreloadScript_addAndRemoveIsNoop_secondRemoval_fails(
         'error': 'no such script',
         'message': f"No preload script with BiDi ID '{bidi_id}'"
     } == exception_info.value.args[0]
+
+
+@pytest.mark.asyncio
+async def test_preload_script_channel(websocket, context_id, html):
+    await subscribe(websocket, "script.message")
+
+    result = await execute_command(
+        websocket, {
+            "method": "script.addPreloadScript",
+            "params": {
+                "functionDeclaration": """(channel) => channel({'foo': 'bar', 'baz': {'1': 2}})""",
```

Comment:
To avoid race condition with the navigation command finished:
```suggestion
                "functionDeclaration": """(channel) => setTimeout(()=>channel({'foo': 'bar', 'baz': {'1': 2}}),1)""",
```